server/release_notes.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"log"
+	"path/filepath"
+	"sync"
+
+	pbVersion "github.com/Percona-Lab/percona-version-service/versionpb/api"
+)
+
+type ReleaseNotes struct {
+	// releaseNotes is a map of PMM versions to release notes.
+	pmmReleaseNotes  map[string]*pbVersion.GetReleaseNotesResponse
+	releaseNotesLock sync.Mutex
+	fs               fs.FS
+}
+
+func NewReleaseNotes(fs fs.FS) *ReleaseNotes {
+	return &ReleaseNotes{
+		pmmReleaseNotes: make(map[string]*pbVersion.GetReleaseNotesResponse),
+		fs:              fs,
+	}
+}
+
+func (r *ReleaseNotes) getVersionsForProduct(product string) map[string]*pbVersion.GetReleaseNotesResponse {
+	switch product {
+	case "pmm":
+		if r.pmmReleaseNotes == nil {
+			r.pmmReleaseNotes = make(map[string]*pbVersion.GetReleaseNotesResponse)
+		}
+		return r.pmmReleaseNotes
+	default:
+		return nil
+	}
+}
+
+func (r *ReleaseNotes) GetReleaseNote(product, version string) (*pbVersion.GetReleaseNotesResponse, error) {
+	r.releaseNotesLock.Lock()
+	defer r.releaseNotesLock.Unlock()
+
+	availableVersions := r.getVersionsForProduct(product)
+	if availableVersions == nil {
+		return nil, errors.New(fmt.Sprintf("%s is not a valid product", product))
+	}
+
+	if notes, ok := availableVersions[version]; ok {
+		return notes, nil
+	}
+
+	log.Println("cache miss")
```

Comment:
What if we print this in debug mode only? I'm afraid it will be too spammy otherwise.

Reply:
Right, left over from debugging )